README.md

```diff
@@ -718,7 +718,16 @@ default version of the stub is Python 3.8.
 provide the path to the tar file in the `EXECUTION_ENV_PATH` in the
 `config.pbtxt` of all the models that want to use the execution environment.
 
-4. If you need to compile the Python backend stub, it is recommended that you
+4. The `EXECUTION_ENV_PATH` **should not** point to anywhere outside the
```

Comment:
It shouldn't matter whether `$$TRITON_MODEL_DIRECTORY` is used or not, because
For this file directory:
```
model_repo
|
`---model_dir
|
`---env_file.tar.gz
```
The user can reference `$$TRITON_MODEL_DIRECTORY/../env_file.tar.gz` or `s3://model_repo/env_file.tar.gz`, which are the same, and both will not work without the patch. Then, the added if statement doesn't cover the latter case, which is inaccurate.

Reply:
This is for eliminating ambiguity, for users that don't know implementation detail, they may think that `$$TRITON_MODEL_DIRECTORY` is just `s3://model_repo/model_dir`, which is exactly the confusion we have seen in the GitHub issue. Under this notion, does that mean `EXECUTION_ENV_PATH: "s3://model_repo/model_dir/../env_file.tar.gz"` is not accepted? This is okay and thus I suggested to add the clause above.

Reply:
Do you mean we divide the responsibility in the following way, for the future:
1. The core only provides access guarantee to the contents **inside** `$$TRITON_MODEL_DIRECTORY`, so paths using `$$TRITON_MODEL_DIRECTORY` should not escape outside the model directory.
2. Any paths that do not include `$$TRITON_MODEL_DIRECTORY` is the responsibility of the backend to fetch the content, whether the content is stored locally or from the cloud.

Then, it means `$$TRITON_MODEL_DIRECTORY/../env_file.tar.gz` path is invalid, as it requires the core to access outside the model directory. `s3://model_repo/env_file.tar.gz` and `s3://model_repo/model_dir/../env_file.tar.gz` are valid from the core's perspective, but it is up to the backend to ultimately decide whether it is valid or not, and the backend has to fetch the file.

All three paths are covered by this patch, so I think we can add the "If $$TRITON_MODEL_DIRECTORY is used in EXECUTION_ENV_PATH, ..." statement, for the future.